### ISSUE TITLE
Error on upgrading to Bokeh v1.4.0 related to custom class in module hierarchy.

### ISSUE THREAD

**Issue by marcuskd** [bokeh/bokeh#9412]
Hi,

I've encountered an error with a custom class after upgrading to Bokeh v1.4.0. I’ve created a minimum example which replicates the problem. The code is below, which needs to be in packages organised as follows:

![cctest](https://user-images.githubusercontent.com/25136779/68809635-4279db80-0664-11ea-8518-74fc7ef07bf5.png)

Custom.py:

```
from bokeh.plotting import Figure
from bokeh.layouts import Column


class Custom(Column):

    __view_model__ = 'Column'
    __subtype__ = 'Custom'

    def __init__(self):

        super().__init__()

        plot = Figure(plot_height=500, plot_width=500)
        plot.circle([1, 2, 3], [3, 2, 1], size=20)

        self.children.append(plot)
```

example.py:

```
from bokeh.plotting import show
from cctest.Custom import Custom

custom = Custom()
show(custom)
```

(The init.py under both cctest and Examples is empty.)

The example runs under Bokeh v1.3.4 but fails under v1.4.0 with the following error:

**Error: Model ‘cctest.Custom.Column’ does not exist. This could be due to a widget or a custom model not being registered before first usage.**

Note that if all the code is in a single module it works fine (see the Discourse topic [here](https://discourse.bokeh.org/t/upgrade-to-bokeh-v1-4-needs-custom-model-registration/4222)) - the issue seems to be the module hierarchy.

**Comment by bryevdv** [comment]
cc @mattpap

**Comment by mattpap** [comment]
A quick workaround is to add a dummy implementation to the model (i.e. `__implementation__ = ""`). This will force old behavior, which doesn't use fully qualified module names.

**Comment by marcuskd** [comment]
@mattpap I just tried that and got the following error:

**RuntimeError: node.js v6.10.0 or higher is needed to allow compilation of custom models ("conda install nodejs" or follow https://nodejs.org/en/download/)**

It's no problem for me to install node.js but does that mean that everyone I distribute my package to would also have to install it?

**Comment by mattpap** [comment]
Right, an empty implementation still triggers compilation and requires nodejs for that. So, the proper solution would be to add `__view_module__ = None` or `__view_module__ = ""`, but that will fail, so another workaround is to use `__view_module__ = "__main__"`.

**Comment by marcuskd** [comment]
Using ```__view_module__ = "__main__"``` worked fine, thanks @mattpap!

**Comment by marcuskd** [comment]
A further error has occurred when running the custom model in the notebook. Code below, with the correction added to address the issue solved above. Needs to be in packages organised as above.

Custom.py:

```
from bokeh.plotting import Figure
from bokeh.layouts import Column


class Custom(Column):

    __view_model__ = 'Column'
    __subtype__ = 'Custom'
    __view_module__ = '__main__'

    def __init__(self):

        super().__init__()

        plot = Figure(plot_height=500, plot_width=500)
        plot.circle([1, 2, 3], [3, 2, 1], size=20)

        self.children.append(plot)
```

example.py:

```
from bokeh.plotting import show
from cctest.Custom import Custom

custom = Custom()
show(custom)
```

example.py runs successfully at the command line. However if I run the following code in the notebook:

```
from bokeh.plotting import show
from bokeh.io import output_notebook
from cctest.Custom import Custom
output_notebook()
custom = Custom()
show(custom)
```

I get the error:
AttributeError: module '__main__' has no attribute '__file__'

Note that if I don't have a heirarchy, i.e. enter the following in the notebook:
```
from bokeh.plotting import Figure, show
from bokeh.layouts import Column
from bokeh.io import output_notebook
output_notebook()

class Custom(Column):

    __view_model__ = 'Column'
    __subtype__ = 'Custom'
    __view_module__ = '__main__'

    def __init__(self):

        super().__init__()

        plot = Figure(plot_height=500, plot_width=500)
        plot.circle([1, 2, 3], [3, 2, 1], size=20)

        self.children.append(plot)

custom = Custom()
show(custom)
```

then it works fine.

Any help appreciated!

Thanks,
Marcus.

**Comment by marcuskd** [comment]
Apologies, closed by accident...

**Comment by marcuskd** [comment]
@mattpap is there a simple workaround I could use to address the notebook problem? Grateful for any suggestions...

**Comment by marcuskd** [comment]
After a quick look at bokeh/bokeh/embed/bundle.py (line 215 onwards) I changed the line:
```__view_module__ = "__main__"```
to
```__view_module__ = "bokeh"```
and found that it now works in the notebook. Is this a suitable workaround or should I be doing something else?

**Comment by mattpap** [comment]
> `__view_module__ = "bokeh"`
and found that it now works in the notebook. Is this a suitable workaround or should I be doing something else?

This is a nice workaround. If `__view_module__` starts with `bokeh`, then fully qualified modules, which cause issues here, will be completely disabled. All this hacking should go await in the upcoming 2.0 release.

**Comment by marcuskd** [comment]
Thanks @mattpap.
